tests/RPS_test.rb

```diff
@@ -0,0 +1,25 @@
+require_relative 'test_helper'
+require_relative '../lib/RockPaperScissors'
+
+class RPSTest <  Minitest::Test
+  def test_assign_play_values
+    x=Game.new
+    x.setup
+    play = "rock"
+    assert_equal(1, x.assign_play_value(play))
+  end
+
+  def test_turn_winner
+    x=Game.new
+    x.setup
+    move1, move2 = 2,3
+    assert_equal(@Player1, x.turn_winner(move1,move2))
```

Comment:
#### Problem 1

At this moment when the tests are running, the value of `@Player1` is definitely `nil`. This probably is not what you intended. Probably you were intended for it to be equal to the name of Player 1. To be more specific, you probably wanted it to be equal to the name of `x`'s Player 1.

But typing `@Player1` here does not actually look inside of `x` to find that property's value. The **only** way to access an object's property/properties is by running a method on the object. So we often have methods for _getting_ a particular property's value, for running a _calculation_ based on some property's value (e.g. `@amount * @tip_percent`), and certainly also for _setting_ a property's value.

So when you've typed `@Player1` here, it's not accessing the information inside of `x`. It's just a random variable out in the wild, with a value of `nil`.

So your assertion thus far is `assert_equal(nil, x.turn_winner(move1,move2))`

**Solution: Write a method in the `Game` class that returns the value of the object's `@Player1` property, and use that method for this part of the assertion.**

#### Problem 2

`turn_winner` requires several pieces of information for it to work correctly:

- move1
- move2
- `@Player1`
- `@Player2`

It needs the moves, because otherwise it can't figure out who won. Those values get passed in as arguments when you call the method.

It needs the Players, because otherwise it's just going to return `nil` regardless of who wins. Those values don't get passed in as arguments–they're supposed to be set ahead of time as instance variables (hence the `@` before the properties).

The problem is that the Players are never defined for `x`. You set up `x` as a sample game, and run the `setup` method to give it some necessary information. But none of the operations you run before the test assertion executes actually give any values to the Players.

So you have `nil` competing against `nil`. Then `nil` plays the Rock and `nil` plays the Paper, and `nil` wins.

Coincidentally, as explained by Problem 1 above, your test compares this result with an orphan `@Player1` (which is also `nil`), and the test passes! :dancer: 

But the problem is that `turn_winner` needs for the Players to be defined first. Otherwise, it can't do it's job.

**Solution: Write a method or two in the `Game` class that sets the values of each Player. Those methods should be run in your test so that `x` has Player values, so when `turn_winner` runs, it has actual Player values.**